pkg/apis/core/types.go

```diff
@@ -2891,6 +2907,16 @@ type PodSecurityContext struct {
 	// sysctls (by the container runtime) might fail to launch.
 	// +optional
 	Sysctls []Sysctl
+	// Defines behavior of changing SELinux labels of the volume before being exposed inside Pod
+	// (Alpha feature).
+	// This field is ignored for Pod's volumes and/or for Linux installations that do not support
+	// SELinux.
+	// Valid values are "OnVolumeMount" and "Always". If not specified, "Always" is used.
+	// "Always" policy recursively changes SELinux labels on all files on all volumes used by the Pod.
+	// "OnVolumeMount" tries to mount volumes used by the Pod with the right context and skip recursive ownership
+	// change. Kubernetes may fall back to policy "Always" if a storage backed does not support this policy.
+	// + optional
+	SELinuxRelabelPolicy *PodSELinuxRelabelPolicy
```

Comment:
Can we group this & `FSGroupChangePolicy` together in a way that makes it clear both pertain to volumes? I'm thinking a sub-struct along the lines of `VolumeSecurityContext`? See above comment, it may make sense to move this to the volume struct.

Reply:
I just followed `fsGroupChangePolicy`, which, after the same discussion, ended up in SecurityContext. @gnufied, do you remember why we kept it there?

Reply:
I would like to know why this wouldn't be in the volume struct too.

Reply:
The discussion from `fsGroupChangePolicy` -  https://github.com/kubernetes/enhancements/pull/1478#discussion_r370904021 . tldr; of that discussion is - 

1. Having this setting on per-volume basis is more complicated for the end user and its puts this burden of deciding which policy to choose for which volume type. 
2. Also, if there were an automated tool which injected such policies in pods, the tool also has to be aware of volumes it should target.
3. On the other hand - having this setting on per-volume basis does not really buy much at the cost of added complexity.